src/Avalonia.Labs.Controls/Cache/ImageCache.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Avalonia.Media.Imaging;
+
+namespace Avalonia.Labs.Controls.Cache
+{
+    /// <summary>
+    /// Provides methods and tools to cache images in a folder
+    /// </summary>
+    public class ImageCache : CacheBase<Bitmap>
+    {
+        /// <summary>
+        /// Private singleton field.
+        /// </summary>
+        [ThreadStatic]
+        private static ImageCache? _instance;
+
+        /// <summary>
+        /// Gets public singleton property.
+        /// </summary>
+        public static ImageCache Instance => _instance ?? (_instance = new ImageCache());
+
+        /// <summary>
+        /// Creates a bitmap from a stream
+        /// </summary>
+        /// <param name="stream">input stream</param>
+        /// <returns>awaitable task</returns>
+        protected override async Task<Bitmap> ConvertFromAsync(Stream stream)
+        {
+            if (stream.Length == 0)
+            {
+                throw new FileNotFoundException();
+            }
+
+            return new Bitmap(stream);
+        }
+
+        /// <summary>
+        /// Creates a bitmap from a cached file
+        /// </summary>
+        /// <param name="baseFile">file</param>
+        /// <returns>awaitable task</returns>
+        protected override async Task<Bitmap> ConvertFromAsync(string baseFile)
+        {
+            using (var stream = File.OpenRead(baseFile))
+            {
+                return await ConvertFromAsync(stream).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Checks whether file is valid or not.
+        /// </summary>
+        /// <param name="file">file</param>
+        /// <param name="duration">cache duration</param>
+        /// <param name="treatNullFileAsOutOfDate">option to mark uninitialized file as expired</param>
+        /// <returns>bool indicate whether file has expired or not</returns>
+        protected override async Task<bool> IsFileOutOfDateAsync(string file, TimeSpan duration, bool treatNullFileAsOutOfDate = true)
+        {
+            if (file == null)
+            {
+                return treatNullFileAsOutOfDate;
+            }
+
+            var fileInfo = new FileInfo(file);
+
+            return fileInfo.Length == 0 || DateTime.Now.Subtract(File.GetLastAccessTime(file)) > duration || DateTime.Now.Subtract(File.GetLastWriteTime(file)) > duration;
```

Comment:
If we want this cache to work on WASM, it should use StorageProvider API.
But unfortunately, there is no way to get cache/internal folder from there. I.e. something like this would be needed https://github.com/AvaloniaUI/Avalonia/issues/10628

Reply:
For now I think current solution is fine

Reply:
Just need to verify that it's not crashing on the browser.